s30272-KamilKownacki/3/zadanie30.txt

```diff
@@ -0,0 +1,136 @@
+Program ps służy do wyświetlania listy procesów. 
+
+Zadania:
+
+A: Zobacz co się stanie jeśli wpiszemy w terminalu:
+
+    ps
+
+    ps -a
+
+    ps x
+
+    ps axu
+    INFO: tu kolejność parametrów a, x, u nie ma znaczenie a jedynie fakt
+, że są podane. Dowolna kolejność powinna dać ten sam wynik.
+
+    Jak myślisz, co oznacza znak zapytania w kolumnie 2?
+    Nie wiesz? A gdzie może być wyjaśnienie?
+
+B: Jak wykonasz poniższe 2 zadania?
+
+    Wyświetl wszystkie procesy bash
+
+    Wyświetl wszystkei procesy należące do użytkownika root
+
+Zadanie A:
+
+$Komenda ps wyświetliła tylko procesy mające ten sam identyfikator użytkownika
+co bieżący użytkownik oraz podłączone do tego samego terminala,
+co użytkownik uruchamiający ps
+$Komenda ps -a wyświetla tylko te procesy, których identyfikator sesji,
+nie jest taki sam jak unikalny identyfikator procesu i nie wyświetla procesów
+spoza aktualnego terminala
+Komenda ps x zdejmuję ograniczenie na wyświetlanie procesów tylko tych
+procesów, które są przypisane do jakiegoś terminala
+$Komenda ps axu wyświetla
+a(Procesy wszystkich użytkowników)
+u(Dodaje dodatkowe szczegóły procesów)
+x(Wypisuje procesy które nie należą do żadnego terminala)
+Ja mam znaki zapytania w 5 kolumnie na ubuntu w kolumnie TTY
+oznacza to identyfikator terminala w którym jest uruchomiony proces
+sprawdzamy oznaczenie terminala poleceniem tty, a co to oznacza przez
+man tty
+
+Zadanie B:
+1.używam
+ps aux | grep bash

```

Comment:
Jeśli będziemy mieć usera o loginie `barnabash` to jego wszystkie procesy zostaną zwrócone co będzie błędnym rozwiązaniem tego zadania. Jak zrobić je poprawnie?